src/main/kotlin/org/rust/lang/core/macros/MacroExpander.kt

```diff
@@ -205,6 +204,33 @@ class MacroExpander(val project: Project) {
                 )
             }
         }
+
+    /** Returns path from [context] to [def]'s crate */
+    private fun getPathToCrate(context: RsElement, def: RsMacro): String {
+        val defCrate = def.crateRoot ?: return ""
+        if (defCrate == context.crateRoot) return ""
+        val (name, relativeDepth) = getRelativeCrateNameAndDepth(context, defCrate) ?: return ""
+        val prefix = when (relativeDepth) {
+            null -> "::"
+            0 -> "self::"
+            else -> "super::".repeat(relativeDepth)
+        }
+        return prefix + name
+    }
+
+    private fun getRelativeCrateNameAndDepth(context: RsElement, externCrateMod: RsMod): Pair<String, Int?>? {
+        val externCrateWithDepth = context.containingMod.superMods.withIndex().mapNotNull { (index, superMod) ->
```

Comment:
I copied this code from autoimport. I can't generalize somehow because autoimports version is too specialized